### PR TITLE
update digital garden feature to use named sanity resources

### PR DESCRIPTION
**Pull request by zacjones93**
This is using a restructured Sanity object where we are querying for the resource title so that we can add resources without breaking the front end.

One thing that I don't understand... `Create a digital garden CLI with Rust` course image and path aren't being returned from the API even though they are present in the query in sanity studio

image

Build the portfolio you need to be a badass web developer   egghead io
Build the portfolio you need to be a badass web developer   egghead io

### CODE REVIEW

File: src/pages/learn/digital-gardening/index.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ const DigitalGardening: React.FC<any> = ({data}) => {
 
 export default DigitalGardening
 
-export const digitalGardeningQuery = groq`*[_type == 'resource' && slug.current == "digital-gardening-for-developers"][0]{
+export const digitalGardeningQuery = groq`*[_type == 'resource' && slug.current == "digital-gardening-for-developers-v2"][0]{
```
Comment (zacjones93) [inline src/pages/learn/digital-gardening/index.tsx:111]: querying v2 for now, slugs have to be unique in sanity

File: src/pages/learn/digital-gardening/index.tsx [hunk 2]
```diff
@@ -122,12 +122,25 @@ export const digitalGardeningQuery = groq`*[_type == 'resource' && slug.current
...
-  resources[]{
+  'featured': resources[title == 'Featured digital gardening courses'][0]{
+ 		'courses': resources[]{
+    	title,
+    	byline,
+      'name': content[title == 'name'][0].description,
+    	'path': resources[]->[0].path,
+    	'image': resources[]->[0].image
+  	}
+  },
```
Comment (zacjones93) [inline src/pages/learn/digital-gardening/index.tsx:133]: This displays weird on github but is formatted as you would expect.

resources is now a query for the resource titled 'Featured digital gardening courses'. Now we can add other sections to the page without the front end automatically rendering what it finds.

File: src/pages/learn/digital-gardening/index.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ const DigitalGardening: React.FC<any> = ({data}) => {
 
         <div>
           <div className="grid lg:grid-cols-12 grid-cols-1 gap-5 mt-5">
-            {data.resources.map((resource: any) => {
+            {data.featured.courses.map((resource: any) => {
```
Comment (zacjones93) [inline src/pages/learn/digital-gardening/index.tsx:66]: rendering featured courses on the page

File: src/pages/learn/digital-gardening/index.tsx [hunk 4]
```diff
@@ -122,12 +122,25 @@ export const digitalGardeningQuery = groq`*[_type == 'resource' && slug.current
...
+  'featured': resources[title == 'Featured digital gardening courses'][0]{
+ 		'courses': resources[]{
+    	title,
+    	byline,
+      'name': content[title == 'name'][0].description,
+    	'path': resources[]->[0].path,
+    	'image': resources[]->[0].image
+  	}
+  },
+	'talks': resources[title == 'Infrastructure for Digital Gardens'][0]{
```
Comment (zacjones93) [inline src/pages/learn/digital-gardening/index.tsx:134]: moved the talks up to the resources section (previously in related resources) and now filtering by resource name

File: src/pages/learn/digital-gardening/index.tsx [hunk 5]
```diff
@@ -80,17 +80,17 @@ const DigitalGardening: React.FC<any> = ({data}) => {
             <div className="flex lg:flex-row flex-col items-center justify-center sm:space-x-10 sm:space-y-0 space-y-5 0 w-full xl:pr-16">
               <div className="mx-auto">
                 <h2 className="text-xs text-yellow-600 dark:yellow-green-300 uppercase font-semibold mb-2 text-center">
-                  {data.related.cta.description}
+                  {data.talks.cta}
```
Comment (zacjones93) [inline src/pages/learn/digital-gardening/index.tsx:83]: similarly iterating over talks

File: src/pages/learn/digital-gardening/index.tsx (hunk 4 above)
Comment (joelhooks) [reply src/pages/learn/digital-gardening/index.tsx:134]: ya this is better imo. i think query by title is brittle and we should use _IDs or slugs

File: src/pages/learn/digital-gardening/index.tsx (hunk 2 above)
Comment (laurosilvacom) [reply src/pages/learn/digital-gardening/index.tsx:133]: This approach makes more sense!

**Comment by zacjones93** [comment]
I'm not sure what issue I was running into but I solved it by deleting the resource reference to the Rust CLI course and adding it back in. It's like the reference was corrupted somehow

File: src/pages/learn/digital-gardening/index.tsx [hunk 6]
```diff
@@ -122,12 +122,25 @@ export const digitalGardeningQuery = groq`*[_type == 'resource' && slug.current
...
+    description,
+    'cta': content[title == 'cta'][0].description,
+		resources[]{
+      title,
+      'path': slug.current,
+      byline,
+      image,
+    },
   },
   related[0]{
```
[the remark this answers is not in the document]
Comment (zacjones93) [reply src/pages/learn/digital-gardening/index.tsx]: I deleted this! good catch, that part of the query was returning no data

**Review by laurosilvacom — APPROVED** [review]

**Pull request merged**